lib/components/mobile/results-screen.js

```diff
@@ -180,54 +97,47 @@ class MobileResultsScreen extends Component {
     // Ensure that narrative covers map.
     narrativeContainerStyle.backgroundColor = 'white'
 
-    let headerAction = null
     const showRealtimeAnnotation = realtimeEffects.isAffectedByRealtimeData && (
       realtimeEffects.exceedsThreshold ||
       realtimeEffects.routesDiffer ||
       !useRealtime
     )
 
-    if (error) {
-      return this.renderError()
-    }
-
     return (
       <MobileContainer>
-        <MobileNavigationBar
-          headerText={resultCount
-            ? `We Found ${resultCount} Option${resultCount > 1 ? 's' : ''}`
-            : 'Waiting...'
-          }
-          headerAction={headerAction}
-        />
-        {this.renderLocationsSummary()}
-
-        <div className='results-map'>
+        <ResultsHeader />
+        <div className={error ? 'results-error-map' : 'results-map'}>
           <Map />
         </div>
-
-        <div
-          className='mobile-narrative-header'
-          style={{ bottom: expanded ? null : 100, top: expanded ? 100 : null }}
-          onClick={this._optionClicked}
-        >
-          Option {activeItineraryIndex + 1}
-          <i className={`fa fa-caret-${expanded ? 'down' : 'up'}`} style={{ marginLeft: 8 }} />
-        </div>
-
-        <div
-          ref='narrative-container'
-          className='mobile-narrative-container'
-          style={narrativeContainerStyle}
-        >
-          <ItineraryCarousel
-            hideHeader
-            expanded={this.state.expanded}
-            onClick={this._optionClicked}
-            showRealtimeAnnotation={showRealtimeAnnotation}
-          />
-        </div>
-        {this.renderDots()}
+        {error
+          ? <ResultsError error={error} />
+          : (
+            <>
+              <div
+                className='mobile-narrative-header'
+                onClick={this._optionClicked}
+                style={{ bottom: expanded ? null : 100, top: expanded ? 100 : null }}
+              >
+                  Option {activeItineraryIndex + 1}
+                <i className={`fa fa-caret-${expanded ? 'down' : 'up'}`} style={{ marginLeft: 8 }} />
```

Comment:
This could be part of a styled component that accepts props and styles accordingly.